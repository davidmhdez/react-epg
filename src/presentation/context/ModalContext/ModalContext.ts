import {createContext} from "react";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type ModalContextValues = {
    isOpen: boolean
    openModal: () => void
    closeModal: () => void
    selectedChannelEvent: ChannelEvent | null
    selectChannelEvent: (channelEvent: ChannelEvent) => void
}

export const initialModalContextValues = {
    isOpen: false,
    selectedChannelEvent: null,
    closeModal: () => {},
    openModal: () => {},
    selectChannelEvent: () => {},
}

const ModalContext = createContext<ModalContextValues>(initialModalContextValues)

export default ModalContext