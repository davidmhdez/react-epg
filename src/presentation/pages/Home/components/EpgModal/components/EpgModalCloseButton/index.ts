export {default} from "./EpgModalCloseButton.tsx"