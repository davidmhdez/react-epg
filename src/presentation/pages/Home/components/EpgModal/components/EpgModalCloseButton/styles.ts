import styled from "styled-components";

export const StyledEpgModalCloseButton = styled.button`
    background: none;
    border: none;
    color: inherit;
    font: inherit;
    cursor: pointer;
    padding: 0;
    margin: 0;
    position: fixed;
    top: 30px;
    right: 30px;
    z-index: 200;
    
    svg{
        width: 30px;
        height: 30px;
        
        path{
            fill: #fff;
        }
    }
`