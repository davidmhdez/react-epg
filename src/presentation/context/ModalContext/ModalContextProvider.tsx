import React, {ReactNode, useState} from "react";
import ModalContext from "./ModalContext.ts";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type Props = {
    children: ReactNode
}

const ModalContextProvider: React.FC<Props> = ({children}) => {
    const [isOpen, setIsOpen] = useState(false);
    const [selectedChannelEvent, setSelectedChannelEvent] = useState<ChannelEvent | null>(null)
    const openModal = () => {
        setIsOpen(true);
        setSelectedChannelEvent(null);
    }
    const closeModal = () => {
        setIsOpen(false);
        setSelectedChannelEvent(null);
    }

    const selectChannelEvent = (channelEvent: ChannelEvent) => setSelectedChannelEvent(channelEvent);

    return (
        <ModalContext.Provider value={{
            isOpen,
            selectedChannelEvent,
            selectChannelEvent,
            openModal,
            closeModal
        }}>
            {children}
        </ModalContext.Provider>
    );
};

export default ModalContextProvider;
