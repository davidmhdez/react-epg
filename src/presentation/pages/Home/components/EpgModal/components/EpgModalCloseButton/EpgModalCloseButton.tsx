import {useContext} from 'react';
import {StyledEpgModalCloseButton} from "./styles.ts";
import ModalContext from "../../../../../../context/ModalContext/ModalContext.ts";
import CloseIcon from "../../../../../../components/icons/CloseIcon.tsx";

const EpgModalCloseButton = () => {
    const { closeModal } = useContext(ModalContext)

    return (
        <StyledEpgModalCloseButton data-testid="closeModalButton" onClick={closeModal}>
            <CloseIcon/>
        </StyledEpgModalCloseButton>
    );
};

export default EpgModalCloseButton;
