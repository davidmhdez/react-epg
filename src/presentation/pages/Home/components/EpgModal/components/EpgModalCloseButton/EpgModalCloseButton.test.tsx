import {fireEvent, render, screen} from "@testing-library/react";
import EpgModalCloseButton from "./EpgModalCloseButton.tsx";
import ModalContext, {initialModalContextValues} from "../../../../../../context/ModalContext/ModalContext.ts";

describe('EpgModalCloseButton test', () => {
    test('should match with snapshot', () => {
        const { container } = render(<EpgModalCloseButton/>);
        expect(container).toMatchSnapshot();
    })

    test('should call closeModal() when click button', ()=>{
        const mockCloseModal = jest.fn();

        render(
            <ModalContext.Provider value={{
                ...initialModalContextValues,
                closeModal: mockCloseModal,
            }}>
                <EpgModalCloseButton/>
            </ModalContext.Provider>
        )

        const button = screen.getByRole('button');
        expect(button).toBeTruthy()
        fireEvent.click(button)
        expect(mockCloseModal).toHaveBeenCalledTimes(1);
    });
})