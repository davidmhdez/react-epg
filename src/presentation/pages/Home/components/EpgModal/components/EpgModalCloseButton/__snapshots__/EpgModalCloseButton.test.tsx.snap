// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpgModalCloseButton test should match with snapshot 1`] = `
<div>
  <button
    class="sc-aYaIB iLwJPY"
    data-testid="closeModalButton"
  >
    <svg
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <mask
        height="24"
        id="mask0_4409_5890"
        maskUnits="userSpaceOnUse"
        width="24"
        x="0"
        y="0"
      >
        <rect
          fill="#D9D9D9"
          height="24"
          width="24"
        />
      </mask>
      <g
        mask="url(#mask0_4409_5890)"
      >
        <path
          d="M6.40008 18.3077L5.69238 17.6L11.2924 12L5.69238 6.4L6.40008 5.6923L12.0001 11.2923L17.6001 5.6923L18.3078 6.4L12.7078 12L18.3078 17.6L17.6001 18.3077L12.0001 12.7077L6.40008 18.3077Z"
          fill="#0F265C"
        />
      </g>
    </svg>
  </button>
</div>
`;
